vms/platformvm/state/state.go

```diff
@@ -1711,6 +1718,165 @@ func (s *state) write(updateValidators bool, height uint64) error {
 	)
 }
 
+func (s *state) writeWeightDiffs(height uint64, weightDiffs map[subnetNodeKey]*ValidatorWeightDiff) error {
+	for k, weightDiff := range weightDiffs {
+		if weightDiff.Amount == 0 {
+			continue
+		}
+
+		err := s.flatValidatorWeightDiffsDB.Put(
+			marshalDiffKey(k.subnetID, height, k.nodeID),
+			marshalWeightDiff(weightDiff),
+		)
+		if err != nil {
+			return err
+		}
+
+		// TODO: Remove this once we no longer support version rollbacks.
+		prefixStruct := heightWithSubnet{
+			Height:   height,
+			SubnetID: k.subnetID,
+		}
+		prefixBytes, err := block.GenesisCodec.Marshal(block.Version, prefixStruct)
+		if err != nil {
+			return fmt.Errorf("failed to create prefix bytes: %w", err)
+		}
+		rawNestedWeightDiffDB := prefixdb.New(prefixBytes, s.nestedValidatorWeightDiffsDB)
+		nestedWeightDiffDB := linkeddb.NewDefault(rawNestedWeightDiffDB)
+
+		weightDiffBytes, err := block.GenesisCodec.Marshal(block.Version, weightDiff)
+		if err != nil {
+			return fmt.Errorf("failed to serialize validator weight diff: %w", err)
+		}
+		if err := nestedWeightDiffDB.Put(k.nodeID[:], weightDiffBytes); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *state) writeBlsKeyDiffs(height uint64, blsKeyDiffs map[ids.NodeID]*bls.PublicKey) error {
+	for nodeID, blsKey := range blsKeyDiffs {
+		key := marshalDiffKey(constants.PrimaryNetworkID, height, nodeID)
+		blsKeyBytes := []byte{}
```

Comment:
Nit. Don't think this actually makes a difference, since we treat `nil`/`[]byte{}` values as being the same, but still.
 
```suggestion
		var blsKeyBytes []byte
```

Reply:
Done